docs/src/main/asciidoc/cdi.adoc

```diff
@@ -0,0 +1,361 @@
+|`@javax.inject.Singleton`   | Just like `@ApplicationScoped` except that no client proxy is used. The instance is created eagerly.
+|`@javax.enterprise.context.RequestScoped`   | The bean instance is associated with the current _request_ (usually an HTTP request). 
+|`@javax.enterprise.context.Dependent`   | This is a pseudo-scope. There is no context object that holds all dependent instances. Instead, a dependent instace is a dependent object of the bean it is injected into.
+|`@javax.enterprise.context.SessionScoped`   | This scope is backed by an `javax.servlet.http.HttpSession` object. It's only available if `quarkus-undertow` extension is used.
+|===
+
+NOTE: There can be other custom scopes provided by Quarkus extension. For example, `qquarkus-narayana-jta` provides `javax.transaction.TransactionScoped`.
+
+[[client_proxies]]
+== _Q: I don't undestand the concept of client proxies._
+
+Indeed, the https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#client_proxies[client proxies, window="_blank"] could be hard to grasp but they provide some useful functionality.
+A client proxy is basically an object that delegates all method invocations to a target bean instance.
+It's a container construct that implements `io.quarkus.arc.ClientProxy` and extends the bean class.
+The client proxy instance is always injected instead of a contextual instance of a normal scoped bean (e.g. `@ApplicationScoped` and `@RequestScoped`).
+Client proxies allow for:
+
+* Lazy instantiation - the instance is created once a method is invoked upon the proxy.
+* Ability to inject a bean with "narrower" scope to a bean with "wider" scope; i.e. you can inject a `@RequestScoped` bean into an `@ApplicationScoped` bean.
+* Circular dependencies in the dependency graph. Circular dependencies are "bad design" but sometimes they're inevitable. 
+* In rare cases it's practical to destroy the beans manually. A direct injected reference would lead to a stale bean instance.
+ 
+ 
+== _Q: OK. You said that there are several kinds of beans?_
+
+A: Yes. In general, in Quarkus we distinguish:
+
+1. Class beans
+2. Producer methods
+3. Producer fields
+4. Synthetic beans
+
+Producer methods are useful if you need additional control over instantiation of a bean.
+They are also useful when integrating third-party libraries where you don't control the class source and may not add additional annotations etc.
+
+.Producer Method Example
+[source,java]
+----
+@ApplicationScoped
+public class PiProducer {
+
+    @Produces <1>
+    double producePi() { 
+       return Math.PI; <2>
+    }
+}
+
+@ApplicationScoped
+public class PiConsumer {
+
+   @Inject
+   double pi; <3>
+   
+   double twoPis() {
+     return pi * 2;
+   }
+} 
+----
+<1> The container analyses the method annotations to build a bean metadata.
```

Comment:
Maybe we should also note that a producer method needs to be declared inside another bean?

Reply:
I wouldn't complicate things here... in quarkus you don't even have to add a bean defining annotation to a class if it declares `@Produces`. I mean it's a good point but why not mention disposers then? Static producers? Etc. WDYT?